cmake/python.cmake

```diff
@@ -15,6 +15,28 @@ if(WIN32)
   file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}" SETUPTOOLS_INSTALL_PREFIX)
 endif()
 
+# a precomputed PYTHON_INSTALL_DIR can be passed along with the matching PYTHON_EXECUTABLE here
+# in order to avoid needing to call PYTHON_EXECUTABLE in the next block in each package
+# the precomputed value is only taken if the PYTHON_EXECUTABLE matches what was found above
+if(DEFINED PYTHON_INSTALL_DIR_TUPLE)
+  string(REPLACE ":" ";" _tuple ${PYTHON_INSTALL_DIR_TUPLE})
+  list(LENGTH _tuple _tuple_len)
+  if(_tuple_len EQUAL 2)
+    list(GET _tuple 0 _tuple_PYTHON_EXECUTABLE)
+    list(GET _tuple 1 _tuple_PYTHON_INSTALL_DIR)
+    if("${_tuple_PYTHON_EXECUTABLE}" STREQUAL "${PYTHON_EXECUTABLE}")
+      set(PYTHON_INSTALL_DIR "${_tuple_PYTHON_INSTALL_DIR}")
```

Comment:
With the current patch this variable is not written to the cache but relies on `PYTHON_INSTALL_DIR_TUPLE` to be in the cache in order to be recomputed every time. Should this also be written to the cache for consistency?